### ISSUE TITLE
unable to re-apply templates with existing resources

### ISSUE THREAD

**Issue by ngamber** [phillbaker/terraform-provider-elasticsearch#86]
Trying to run a second apply with resources that already exist in the cluster and get failures like this: 

```
 : elastic: Error 409 (Conflict): [test1]: version conflict, document already exists (current version [1]) [type=version_conflict_engine_exception]

  on elasticsearch.tf line 21, in resource "elasticsearch_opendistro_ism_policy" "test1":
  21: resource "elasticsearch_opendistro_ism_policy" "test1" {



Error: elastic: Error 400 (Bad Request): index_template [test1] already exists [type=illegal_argument_exception]

  on elasticsearch.tf line 43, in resource "elasticsearch_index_template" "test1":
  43: resource "elasticsearch_index_template" "test1" {
```

This happens whether the policy is provided in-line or as a body= file() reference. Not sure if there's a better way to handle this but not seeing a difference in the examples.

**Comment by ngamber** [comment]
Deleting the resources and then letting terraform create them again seems to fix this.

**Comment by tdmalone** [comment]
@ngamber I'm experiencing this problem too, with the same resource. Deleting and re-creating might be a workaround (assuming Elasticsearch allows you to delete an ISM policy that's already attached to an index - I haven't got that far yet), but it's not really a solution. Are you happy to reopen this issue?

**Comment by tdmalone** [comment]
Actually - now that I've applied the policy to an index, I _can_ make updates to it in Terraform. So possibly this is an issue with unapplied policies only? In which case - I guess the workaround works, but it's still not quite how Terraform is meant to work.

**Comment by ngamber** [comment]
That was our "workaround," to create them without using this provider, then use it to update them after the fact.

**Comment by phillbaker** [comment]
@ngamber @tdmalone can you include a simple terraform configuration file that reproduces this issue?

**Comment by phillbaker** [comment]
If the root cause of the failure here is the HTTP 409 on the ISM policy, it seems similar to https://github.com/phillbaker/terraform-provider-elasticsearch/issues/154, where we'll need to implement a retry in the provider to handle version conflicts. That issue was also worked around by using a `-parallelism=1`, @ngamber @tdmalone can either of you try that flag to confirm whether that solution is effective?

**Comment by phillbaker** [comment]
I believe this should be fixed in 2c624964a6939abbe604b266a1f19aa0991c5cd4.

**Comment by tdmalone** [comment]
Apologies for the delay @phillbaker and thanks for looking at this.

I'm no longer able to replicate this issue on 1.5.5, 1.5.4, 1.5.3, 1.5.2, or even 1.4.4 (version before the last changes to ISM in 1.5.0), so all I can guess is that it was perhaps an upstream API issue that has since been resolved?

(I've come across another small issue with this resource, but I'll create a separate issue for that).
